sources/editor/Xenko.Assets.Presentation/AssetEditors/Gizmos/TransformationGizmo.cs

```diff
@@ -434,6 +504,27 @@ private async Task TransformSceneEntityBase()
             }
         }
 
+        protected Vector3 GetScaledLocation(Vector3 position, Vector3 scaleOrigin, Vector3 scale)
+        {
+            var vec = new Vector3();
+            vec.X = GetScaledAxis(position.X, scaleOrigin.X, scale.X);
+            vec.Y = GetScaledAxis(position.Y, scaleOrigin.Y, scale.Y);
+            vec.Z = GetScaledAxis(position.Z, scaleOrigin.Z, scale.Z);
+            return vec;
+        }
+
+        protected float GetScaledAxis(float position, float scaleOrigin, float scale)
+        {
+            return scaleOrigin + (position - scaleOrigin) * scale;
+        }
+
+        protected Vector3 Transform(Vector3 vector, Matrix transform)
```

Comment:
The only Transform without refs returns a Vector4. I could either put a new method in Vector3 called something like `TransformVec3` or convert the Vector4 to Vector3, not sure which one is better.

Reply:
I don't really see a reason to implement that function, you could use ``Transform(ref initialTranslation, ref inverseScaleSpace, out var position);``, on line 484 and others, I don't see a problem with that ?
The vector API uses refs to pass matrix and vectors more efficiently, passing large structs as method parameters incur a fairly substantial cost if not done by reference.

Reply:
I didn't think the performance hit would be significant enough and it seemed to be consistent with the functions around along with the stack of local variables I had earlier.

Reply:
In that case, should `GetScaledLocation` use refs as well?

Reply:
Nah you can keep using non-refs, you don't have to force yourself, I was just clarifying why the API guides you towards using refs. The cost is just large enough to warrant using an already existing ref function instead of its normal counterpart if the readability of your code does not suffer from it, which, in that case, seemed like a perfect fit.